bindings/src/test/java/io/pravega/storage/filesystem/FileSystemMockTests.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.storage.filesystem;
+
+import io.pravega.segmentstore.storage.SegmentHandle;
+import lombok.Getter;
+import lombok.Setter;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.mockito.ArgumentCaptor;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.channels.FileChannel;
+import java.nio.channels.spi.AbstractInterruptibleChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class FileSystemMockTests {
```

Comment:
Why didn't you put these tests in the `FileSystemStorageTests.java` class?

Reply:
Because mock based tests are testing in a fundamentally different way. We need to inherit from FileSystemStorage and setup mock and calls every explicitly. Therefore keeping them in a separate file. (Plus I'll be adding more mock based tests later.)